core/imageroot/usr/local/agent/pypkg/cluster/grants.py

```diff
@@ -99,3 +99,37 @@ def alter_user(rdb, user, revoke, role, on_clause):
 
     pipe.execute()
 
+def refresh_permissions(rdb):
+    """Scan the stored authorizations and re-apply permissions to all modules"""
+    for kauth_module in rdb.scan_iter('cluster/authorizations/module/*'):
+        kmid = kauth_module.removeprefix('cluster/authorizations/module/')
+        authorizations = rdb.smembers(kauth_module) or []
+        mnode_id = rdb.hget('cluster/module_node', kmid)
+        add_module_permissions(rdb, kmid, authorizations, mnode_id)
+
+def add_module_permissions(rdb, module_id, authorizations, node_id):
+    """Parse authorizations and grant permissions to module_id"""
+    for authz in authorizations:
+        if '*' in authz:
+            print(agent.SD_WARNING + f'Cannot grant authorization "{authz}" to {module_id}. Wildcard is not allowed, use @any suffix instead.')
+            continue
+
```

Comment:
If wildcard is not allowed, why even bother?
```suggestion
```

Reply:
I don't understand the code suggestion. The `for` statement can't be removed!

The whole function body is a cut-paste from the previous add-module action implementation.

I'm removing the `print` statement anyway because it is not good in a library function.

Reply:
I selected the wrong line.
I'm proposing to delete lines from 113 to 115

Reply:
I moved those lines to a point where the validation rule can be enforced

See commit https://github.com/NethServer/ns8-core/pull/321/commits/4dcb19936290277aadf1c2f056091f2e4b719190